tools/wake/main.cpp

```diff
@@ -713,7 +713,7 @@ int main(int argc, char **argv) {
 
   if (!quiet && alerted_slow_cache && is_stdout_tty) {
     std::cout << "Scanning " << wakefilenames.size() << "/" << wakefilenames.size()
-              << " wake files. Kernel file cache may be cold." << std::endl;
+              << " wake files.\r" << std::endl;
```

Comment:
Ashley can definitely give a better answer, but I think this deliberately *didn't* have the `\r` because it was meant to stay on the screen as a record of "yeah, we took a long time to scan everything, but we did find them eventually!"  (Note the `wakefilenames.size() << "/" << wakefilenames.size()` having the total number on both sides.)

Reply:
Now that you say it, I remember recommending this or backspaces and I remember him not wanting to do it for some reason.

Reply:
If I run `wake -x Unit` on a slow machine, I should be left with nothing but `Unit`, but I see
```
$ wake -x Unit                                                                                                                                                   
Finding wake files is taking longer than expected. Kernel file cache may be cold. (45753 explored).                                                                                            
Scanning 267/267 wake files. Kernel file cache may be cold.                                                                                                                                    
Unit
```

Reply:
I'm not sure we've guaranteed that (unused symbol warnings will still be printed, for example, though that might be to stderr?), but I do like it in principle.

Reply:
That only happens when you output to a terminal. If we used your solution without a terminal you'd still have bad bytes so I don't think this is an issue.

Reply:
(plying devil's advocate of course, I'm happy with using \r unconditionally here).

Reply:
@V-FEXrt do you have any thoughts on these changes?

Reply:
Hi sorry, was heads down in code today. Looking at the review now but my personal preference was to not erase. (but IIRC it was a preference and nothing else)

Reply:
Okay, yep. I'm happy with clearing. But can you make this line just a newline then?

`std::cout << "Scanning " << wakefilenames.size() << "/" << wakefilenames.size() << " wake files.\r" << std::endl;`
              
`\r` following by `std::endl` is just the same as only emitting the `std::endl`


Also FWIW, we only print this when `is atty = true` so scripts _shouldn't _ see the output only humans